packages/circuit-ui/components/AutocompleteInput/AutocompleteInput.tsx

```diff
@@ -0,0 +1,494 @@
+      [onChange],
+    );
+
+    const onComboboxChange = useCallback(
+      (event: ChangeEvent<HTMLInputElement>) => {
+        setSearchText(event.target.value);
+        setActiveSuggestion(undefined);
+        if (event.target.value.length >= minQueryLength) {
+          debouncedOnChange?.(event);
+        }
+      },
+      [minQueryLength, debouncedOnChange],
+    );
+
+    const onComboboxClear = useCallback(
+      (event: ClickEvent) => {
+        changeInputValue(textBoxRef.current, '');
+        onClear?.(event);
+      },
+      [onClear],
+    );
+
+    const onPresentationFieldClear = useCallback(
+      (event: ClickEvent) => {
+        setPresentationFieldValue('');
+        setSearchText('');
+        changeInputValue(presentationFieldRef.current, '');
+        setIsOpen(true);
+        onClear?.(event);
+      },
+      [onClear],
+    );
+
+    const onPresentationFieldClick = useCallback(() => {
+      setIsOpen(true);
+    }, []);
+
+    const onComboboxClick = useCallback(() => {
+      textBoxRef?.current?.select();
+      setIsOpen(true);
+    }, []);
+
+    const { floatingStyles, refs, update } = useFloating<HTMLElement>({
+      open: isOpen,
+      placement: 'bottom',
+      strategy: 'fixed',
+      middleware: [
+        offset(8),
+        shift({ padding: boundaryPadding }),
+        flip({ padding: boundaryPadding, fallbackPlacements: ['top'] }),
+        size(sizeOptions),
+      ],
+    });
+
+    const onSuggestionClick = useCallback(
+      (selectedValue: string) => {
+        onSelection(selectedValue);
+        closeSuggestionBox();
+      },
+      [onSelection, closeSuggestionBox],
+    );
+
+    useEffect(() => {
+      if (!isOpen && searchText !== value) {
+        changeInputValue(textBoxRef.current, getSuggestionLabel(value));
+        if (isImmersive) {
+          setPresentationFieldValue(getSuggestionLabel(value));
+        }
+      }
+    }, [isOpen, isImmersive, value, searchText, getSuggestionLabel]);
+
+    const onInputKeyDown: KeyboardEventHandler<HTMLInputElement> = useCallback(
+      (event) => {
+        if (isOpen) {
+          if (isArrowDown(event) || isArrowUp(event)) {
+            const totalShownSuggestions =
+              resultsRef.current?.querySelectorAll('[role="option"]').length ??
+              0;
+            event.preventDefault();
+
+            if (activeSuggestion === undefined) {
+              setActiveSuggestion(
+                isArrowDown(event) ? 0 : suggestionValues.length - 1,
+              );
+            } else {
+              const nextSuggestion =
+                (activeSuggestion +
+                  totalShownSuggestions +
+                  (isArrowDown(event) ? 1 : -1)) %
+                totalShownSuggestions;
+
+              setActiveSuggestion(nextSuggestion);
+            }
+          }
+          if (isEnter(event) && activeSuggestion !== undefined) {
+            onSuggestionClick(suggestionValues[activeSuggestion]);
+          }
```

Comment:
I believe we should call `event.preventDefault();` to prevent accidental form submissions.